Assets/GameAssets/Player/PlayerScripts/PlayerReticle.cs

```diff
@@ -52,22 +56,44 @@ private void Start()
         ObtainReferences();
 
         InitializeReticle();
+
+        _isFocusChanging = false;
     }
 
     /// <summary>
-    /// Checks for inputs related to the harpoon gun and appropriately updates UI.
+    /// Called by the harpoon when the state of the reticle needs to change (focusing and unfocusing)
     /// </summary>
-    private void Update()
+    public void ChangeFocus()
     {
-        // Updates the sprite of the reticle if spread range is actively changing due to focus
-        if (_harpoonGunScript.GetCurrentFocusState() == HarpoonGun.EFocusState.Focusing ||
-            _harpoonGunScript.GetCurrentFocusState() == HarpoonGun.EFocusState.Unfocusing)
-        {
-            AdjustReticleSize();
-        }
+        _isFocusChanging = true;
+        StartCoroutine(ChangingReticleFocus());
+    }
 
-        AdjustReticleAppearance();
+    /// <summary>
+    /// Used to update the reticle appearance and size
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator ChangingReticleFocus()
+    {
+        while (_isFocusChanging)
+        { 
+            if (_newReticleSize < _maxScopeSize || _newReticleSize > _minScopeSize)
+            {
+                yield return new WaitForFixedUpdate();
+                AdjustReticleSize();
+                AdjustReticleAppearance();
+            }
+            else
+            {
+                _isFocusChanging = false;
+            }
+        }
+    }
 
+        public void ReticleFire()
+    {
+        //_newReticleSize = _maxScopeSize;
```

Comment:
why is this commented out? if this is being scrapped later delete for now then if need be you can bring back with github history or uncomment if not intended.